docs/guidance_for_upgraders.md

```diff
@@ -0,0 +1,74 @@
+# Guidance for Operator Developer
+
+After you change either the operator signature in a BC breaking way or the semantics of the operator, you will need to write an “upgrader” to make the change non-BC breaking iff they are used in TorchScript or mobile. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `
+
+The steps to write upgrader:
+
+1. Prepare a model before making changes to the operator, following the process:
+    1. Add a module in `test/jit/fixtures_srcs/fixtures_src.py`. The reason why we generate the model file before updating the operator is that, an old model with the old operator before the change is needed to ensure the upgrader is working as expected. In `test/jit/fixtures_srcs/generate_models.py`, add the module and its corresponding changed operator name like following:
+
+    ```
+    ALL_MODULES = {
+        TestVersionedDivTensorExampleV7(): "aten::div.Tensor",
+    }
+    ```
+
+    This module should include the changed operator. If the operator isn't covered in the model, the model export process will fail.
+
+    2. Export the model to `test/jit/fixtures` by running
+    ```
+    python test/jit/fixtures_src/generate_models.py
+    ```
+    3. Commit the change and submit a PR
+
+2. Write an upgrader in `torch/csrc/jit/operator_upgraders/upgraders_entry.cpp` file inside a map `kUpgradersEntryMap`. The softly enforced naming format is `<operator_name>_<operator_overload>_<start>_<end>`. For example, the below example means that `linspace.out `and` linspace` at versions from 0 to 7 need to be replaced by these upgraders. Note that the upgrader is essentially a python source code that is TorchScriptable.
```

Comment:
> For example, the below example means that `linspace.out `and` linspace` at versions from 0 to 7 need

I think this is technically correct, but there is no instruction on how the developer could find out what `0` and `7` should be.